### DISCUSSION TITLE
Small world, Sigma and Omega

### DISCUSSION THREAD

**Discussion by 1101AlexZab1011** [networkx/networkx#4725] [Q&A]
Hello

I try to compute networkx.algorithms.smallworld.omega and networkx.algorithms.smallworld.sigma for graph with 75 nodes and 1249 edges, but it takes a very long time to compute. Is there some way to make it faster?

Thanks

**Comment by dschult** [comment]
How long is very long?   I think my desktop machine can do that calculation in a couple of hours.

`omega` and `sigma` are computationally demanding quantities. I'm assuming you are not running out of RAM (with only 1K edge you would need a lot of edge attributes or many many copies of the graph to run out of memory).  

It looks like (trials with binomial graph p=0.5 and n increasing by factors of 2) that `nx.sigma` take O(n^4) time to compute.
You can of course adjust the parameters to decrease `niter` and `nrand` so the number of random variants computed is less. cutting both to half their default value cuts the time by a factor of 3. 

The basic problem with `sigma` and `omega` is that they use average shortest path length which requires finding the shortest path between every pair of vertices. Then you have to repeat this for many representative graphs.

**Reply by 1101AlexZab1011** [reply]
I interrupted the program after five hours of work. Probably I have too weak machine for it
Thank you for answer, I'll try to decrease niter and nrand
